### PR TITLE
make default FW approach land speed the same as Qplane

### PR DESCRIPTION
**Pull request by Hwurzburg**
if TECS_LAND_ARSPD = -1 make it use halfway between trim airspeed and airspeed min, like Qplane does for its fixed wing approach

this also makes sure that RTLs are at trim cruise in Qplane until approach begins....this is a minor change in behaviour...(actually back to older behaviour )but I think a good one

### CODE REVIEW

File: libraries/AP_Landing/AP_Landing_Slope.cpp [hunk 1]
```diff
@@ -351,7 +351,10 @@ int32_t AP_Landing::type_slope_get_target_airspeed_cm(void)
     case SlopeStage::APPROACH:
         if (land_airspeed >= 0) {
             target_airspeed_cm = land_airspeed * 100;
-        }
+        } else {
+            const float cruise_speed = aparm.airspeed_cruise_cm * 0.01;
+            target_airspeed_cm = 0.5 * (cruise_speed + aparm.airspeed_min);
```
Comment (IamPete1) [inline libraries/AP_Landing/AP_Landing_Slope.cpp:356]:
```suggestion
            target_airspeed_cm = 0.5 * (aparm.airspeed_cruise_cm + aparm.airspeed_min * 100);
```

Comment (Hwurzburg) [reply libraries/AP_Landing/AP_Landing_Slope.cpp]: thanks

File: libraries/AP_Landing/AP_Landing_Slope.cpp [hunk 2]
```diff
@@ -351,7 +351,9 @@ int32_t AP_Landing::type_slope_get_target_airspeed_cm(void)
     case SlopeStage::APPROACH:
         if (land_airspeed >= 0) {
             target_airspeed_cm = land_airspeed * 100;
-        }
+        } else {
+            target_airspeed_cm = 0.5 * (aparm.airspeed_cruise_cm + aparm.airspeed_min * 100);
```
Comment (tridge) [inline libraries/AP_Landing/AP_Landing_Slope.cpp:355]: this misses the one in mode_auto_target_airspeed_cm()

Comment (tridge) [reply libraries/AP_Landing/AP_Landing_Slope.cpp:355]: we need a single function in AP_Landing for get_landing_airspeed() in m/s, then used in all places

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_Landing/AP_Landing_Slope.cpp (hunk 2 above)
Comment (Hwurzburg) [reply libraries/AP_Landing/AP_Landing_Slope.cpp]: revised per above...

File: ArduPlane/quadplane.cpp [hunk 3]
```diff
@@ -4114,23 +4114,16 @@ float QuadPlane::get_land_airspeed(void)
...
-            approach_speed = tecs_land_airspeed;
-        } else {
-            if (qstate == QPOS_APPROACH) {
-                // default to half way between min airspeed and cruise
-                // airspeed when on the approach
-                approach_speed = 0.5*(cruise_speed+plane.aparm.airspeed_min);
-            } else {
-                // otherwise cruise
+        float approach_speed = plane.TECS_controller.get_land_airspeed();    
+        if (qstate != QPOS_APPROACH) {
```
Comment (IamPete1) [inline ArduPlane/quadplane.cpp:4118]: This logic is not quite the same. Previously we would always use the `LAND_ARSPD` parameter if set. Now we only use it if not in approach. 

Unfortunately I don't see a good way to fix it.

Maybe the change in behavior is OK in any-case.

Comment (Hwurzburg) [reply ArduPlane/quadplane.cpp:4118]: it still uses it if its set...look at the get_land_airspeed function

Comment (IamPete1) [reply ArduPlane/quadplane.cpp:4118]: Previously `tecs_land_airspeed` (`TECS_LAND_ARSPD`) was always used if positive. Now we only use it if positive AND not in approach.

So now in approach we return at `cruise_speed` and not `tecs_land_airspeed`.

There is a change in behaviour. 

As I say, I'm not sure if its good or bad.

Comment (Hwurzburg) [reply ArduPlane/quadplane.cpp:4118]: sorry, it is a minor additional change (back to original RTL speed behaviour before the change to interpolated landing airspeed changed it for all of the RTL)...but a good one...

File: ArduPlane/quadplane.cpp [hunk 4]
```diff
@@ -4114,23 +4114,16 @@ float QuadPlane::get_land_airspeed(void)
     if (qstate == QPOS_APPROACH ||
         plane.control_mode == &plane.mode_rtl) {
```
Comment (tridge) [inline ArduPlane/quadplane.cpp:4115]: this was supposed to be qrtl not rtl, it can never be in rtl when this function is called (existing bug)

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by Hwurzburg** [comment]
Closing in favor of #23601

**Pull request closed without merging**